gateway/src/apicast/policy/find_service/find_service.lua

```diff
@@ -1,70 +1,23 @@
-local Policy = require('apicast.policy')
-local _M = Policy.new('Find Service Policy')
 local configuration_store = require 'apicast.configuration_store'
-local mapping_rules_matcher = require 'apicast.mapping_rules_matcher'
-local new = _M.new
-
-local function find_service_strict(configuration, host)
-  local found
-  local services = configuration:find_by_host(host)
-
-  for s=1, #services do
-    local service = services[s]
-    local hosts = service.hosts or {}
-
-    for h=1, #hosts do
-      if hosts[h] == host and service == configuration:find_by_id(service.id) then
-        found = service
-        break
-      end
-    end
-    if found then break end
-  end
-
-  return found or ngx.log(ngx.WARN, 'service not found for host ', host)
-end
-
-local function find_service_cascade(configuration, host)
-  local found
-  local services = configuration:find_by_host(host)
-  local method = ngx.req.get_method()
-  local uri = ngx.var.uri
-
-  for s=1, #services do
-    local service = services[s]
-    local hosts = service.hosts or {}
+local host_based_finder = require('apicast.policy.find_service.host_based_finder')
+local path_based_finder = require('apicast.policy.find_service.path_based_finder')
 
-    for h=1, #hosts do
-      if hosts[h] == host then
-        local name = service.system_name or service.id
-        ngx.log(ngx.DEBUG, 'service ', name, ' matched host ', hosts[h])
-
-        local matches = mapping_rules_matcher.matches(method, uri, {}, service.rules)
-        -- matches() also returns the index of the first rule that matched.
-        -- As a future optimization, in the part of the code that calculates
-        -- the usage, we could use this to avoid trying to match again all the
-        -- rules before the one that matched.
-
-        if matches then
-          found = service
-          break
-        end
-      end
-    end
-    if found then break end
-  end
+local Policy = require('apicast.policy')
+local _M = Policy.new('Find Service Policy')
 
-  return found or find_service_strict(configuration, host)
-end
+local new = _M.new
 
 function _M.new(...)
   local self = new(...)
 
   if configuration_store.path_routing then
     ngx.log(ngx.WARN, 'apicast path routing enabled')
-    self.find_service = find_service_cascade
+    self.find_service = function(configuration, host)
```

Comment:
This function can be defined in a local variable when loading the file, right?

Reply:
That would be problematic to test as you mentioned above.
This only runs when the configuration is loaded, so it should be fine.

Reply:
That would be problematic to test as you mentioned above.
This only runs when the configuration is loaded, so it should be fine.